src/Statement/StatementList.php

```diff
@@ -282,4 +282,24 @@ public function isEmpty() {
 		return empty( $this->statements );
 	}
 
+	/**
+	 * @since 3.0
+	 *
+	 * @param Statement $statement
+	 *
+	 * @return int|bool
+	 */
+	public function indexOf( Statement $statement ) {
+		$index = 0;
+
+		foreach ( $this->statements as $s ) {
+			if ( $s->equals( $statement ) ) {
```

Comment:
Nope. Was just an example, can also use different ranks as an example. I still think implementations like `indexOf`, `contains` and `remove` should work with the GUID, not with `equals`.